utils/state.go

```diff
@@ -25,6 +25,12 @@ type IOStateManager struct {
 
 // ReadState will de-serialize the secrets state from JSON format storage on disk.
 func (ism IOStateManager) ReadState() (Secrets, error) {
+	_, err := os.Stat(ism.filename)
+	if os.IsNotExist(err) {
+		// The file doesn't exist, dont raise an error and return empty data to account
+		// for first-run without a state serialized to disk.
+		return []Secret{}, nil
+	}
```

Comment:
Not critical since we are immediately reading after this, so any error should also be made evident there, but we could check for non NotExist error and return on that, e.g.

```go
if err != nil {
    if os.IsNotExist(err) {
        return []Secret{}, nil
    }
    return nil, err
}
```

Reply:
Waiting on CI to pass before merging, but adopted this nested return vs the block at the top of the method. :+1: